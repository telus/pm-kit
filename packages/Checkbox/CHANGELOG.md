# Change Log

All notable changes to this project will be documented in this file.
See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.

## [1.1.3](https://github.com/telus/pm-kit/compare/@pm-kit/checkbox@1.1.2...@pm-kit/checkbox@1.1.3) (2020-01-22)

**Note:** Version bump only for package @pm-kit/checkbox





## [1.1.2](https://github.com/telus/pm-kit/compare/@pm-kit/checkbox@1.1.1...@pm-kit/checkbox@1.1.2) (2020-01-16)


### Bug Fixes

* **deps:** update dependency framer-motion to v1.7.0 ([#21](https://github.com/telus/pm-kit/issues/21)) ([9381a7d](https://github.com/telus/pm-kit/commit/9381a7d34c314af7f6aec1e32b322e6e4f104c02))





## [1.1.1](https://github.com/telus/pm-kit/compare/@pm-kit/checkbox@1.1.0...@pm-kit/checkbox@1.1.1) (2019-12-10)


### Bug Fixes

* **other:** fixed npmignore pattern for stories, updated changelogs ([0e02d0c](https://github.com/telus/pm-kit/commit/0e02d0c53b3a88905d51d4a8cc1b7e8f6da939fa))





# [1.1.0](https://github.com/telus/pm-kit/compare/@pm-kit/checkbox@1.0.0...@pm-kit/checkbox@1.1.0) (2019-12-10)

Added .npmignore file to remove certain files from published package

# 1.0.0 (2019-12-06)

### Features

- **checkbox:** added Checkbox package ([#11](https://github.com/telus/pm-kit/pull/11)) ([0bee47e](https://github.com/telus/pm-kit/commit/0bee47e61af6d2b10fd777f244492bc8ce787fc6))
