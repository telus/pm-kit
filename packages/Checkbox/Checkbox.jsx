import React, { forwardRef } from 'react'
import PropTypes from 'prop-types'
import { css } from '@emotion/core'
import { parkGreen, red } from '../colours/colours'
import { motion, AnimatePresence } from 'framer-motion'

/**
 * @version ./package.json
 */

const container = css`
  display: flex;
  align-items: center;
  width: 100% !important;

  height: 100% !important;

  left: 0 !important;

  top: 0 !important;
`

const fakeCheckbox = css`
  height: 15px;
  width: 15px;
  min-height: 20px;
  min-width: 20px;
  outline: 0;
  line-height: 0;
  display: flex;
  align-items: center;
  justify-content: center;
  cursor: pointer;
  margin-top: 0.125rem;
  border: 2px solid ${parkGreen};
  border-radius: 6px;
`

const checkedFakeCheckbox = css`
  height: 15px;
  width: 15px;
  min-height: 20px;
  min-width: 20px;
  outline: 0;
  line-height: 0;
  display: flex;
  align-items: center;
  justify-content: center;
  cursor: pointer;
  margin-top: 0.125rem;
  border: 2px solid ${parkGreen};
  border-radius: 6px;
  background-color: ${parkGreen};
`

const fakeCheckboxInner = css`
  & > svg {
    & path {
      stroke: white;
    }
  }
`

const hiddenInput = css`
  position: fixed;
  opacity: 0;
  pointer-events: none;
`

const image = css`
  height: 15px;
  width: 15px;
`

const feedbackError = css`
  font-weight: 500;
  color: ${red};
`
const styledLabel = css`
  display: flex;
`

const labelText = css`
  font-weight: bold;
  margin-left: 12px;
`

const label_ = css`
  display: flex;
  flex-wrap: wrap;
  margin: 0 0 0 52px;
  font-size: 14px;
  & label {
    letter-spacing: 0.05px;
    margin-right: 0.5rem;
  }
`

const checkVariants = {
  hidden: { scale: 0 },
  show: { scale: 0.6 },
  transition: {
    duration: 0.5,
  },
}

export const Checkbox = (
  { error, feedback, label, name, value, id, onChange, checked, forwardedRef, ...rest },
  ref
) => {
  const styles = {
    labelStyles: label_,
    hiddenInputStyles: hiddenInput,
    styledLabelStyles: styledLabel,
    containerStyles: container,
    fakeCheckboxStyles: fakeCheckbox,
    checkedFakeCheckboxStyles: checkedFakeCheckbox,
    fakeCheckboxInnerStyles: fakeCheckboxInner,
    labelTextStyles: labelText,
    feedbackErrorStyles: feedbackError,
    imageStyles: image,
  }
  //   const inputId = id || getGeneratedId(name, value)
  const renderFeedback = errorMessage => <span css={styles.feedbackErrorStyles}>{`(${errorMessage})`}</span>
  return (
    <AnimatePresence>
      <div {...rest}>
        <div css={styles.labelStyles}>{feedback === 'error' && error && renderFeedback(error)}</div>
        <input
          css={styles.hiddenInputStyles}
          type="checkbox"
          value={value}
          id={1}
          name={name}
          onChange={onChange}
          checked={checked}
          ref={forwardedRef}
        />
        <label css={styles.styledLabelStyles} htmlFor={1}>
          <div css={styles.containerStyles}>
            <span css={checked ? styles.checkedFakeCheckboxStyles : styles.fakeCheckboxStyles}>
              <motion.span
                css={styles.fakeCheckboxInnerStyles}
                variants={checkVariants}
                animate={checked ? 'show' : 'hidden'}
                exit={{ opacity: 0 }}
              >
                <svg xmlns="http://www.w3.org/2000/svg" width="20" height="17">
                  <path
                    fill="none"
                    fillRule="evenodd"
                    stroke="#034045"
                    strokeWidth="4"
                    d="M18.031 1.866L7.154 13.442l-5.37-5.944"
                  />
                </svg>
              </motion.span>
            </span>

            <span css={styles.labelTextStyles}>{label}</span>
          </div>
        </label>
      </div>
    </AnimatePresence>
  )
}

Checkbox.propTypes = {
  error: PropTypes.string,
  feedback: PropTypes.oneOf(['success', 'error']),
  label: PropTypes.string.isRequired,
  name: PropTypes.string.isRequired,
  value: PropTypes.oneOfType([PropTypes.string, PropTypes.number, PropTypes.bool]).isRequired,
  id: PropTypes.string,
  onChange: PropTypes.func,
  checked: PropTypes.bool,
}

Checkbox.defaultProps = {
  id: undefined,
  error: undefined,
  feedback: undefined,
  onChange: undefined,
  checked: undefined,
  label: '',
  name: '',
  value: '',
}

const CheckboxWithRef = forwardRef((props, ref) => <Checkbox {...props} forwardedRef={ref} />)

export default CheckboxWithRef
