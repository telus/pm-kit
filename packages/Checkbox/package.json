{
  "name": "@pm-kit/checkbox",
  "version": "1.0.0-0",
  "main": "dist/index.cjs.js",
  "module": "dist/index.es.js",
  "license": "MIT",
  "publishConfig": {
    "access": "public"
  },
  "author": "Public Mobile",
  "engines": {
    "node": ">=8"
  },
  "bugs": {
    "url": "https://github.com/telus/public-mobile-design-system/issues"
  },
  "peerDependencies": {
    "@emotion/core": "^10.0.22",
    "react": "^16.12.0",
    "react-dom": "^16.12.0"
  },
  "dependencies": {
    "prop-types": "^15.7.2",
    "framer-motion": "1.6.17"
  }
}
