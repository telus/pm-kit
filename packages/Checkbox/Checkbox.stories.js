import React, { useState } from 'react'
import Checkbox from './Checkbox'
import { withKnobs, select, text } from '@storybook/addon-knobs'

export default {
  title: 'Design System|Checkbox',
  component: Checkbox,
  decorators: [withKnobs],
}

export const Default = () => {
  const [checked, setChecked] = useState(true)

  const updateCheck = () => {
    setChecked(!checked)
  }
  return (
    <Checkbox
      labelName="Testing"
      name="Test"
      value={true}
      onChange={updateCheck}
      checked={checked}
      error="This field is required"
    />
  )
}

export const CheckboxWithError = () => {
  const [checked, setChecked] = useState(false)

  const updateCheck = () => {
    setChecked(!checked)
  }
  return (
    <Checkbox
      labelName="Testing"
      name="Test"
      value={true}
      onChange={updateCheck}
      checked={checked}
      feedback={checked ? undefined : 'error'}
      error="This field is required"
    />
  )
}

export const Playground = () => {
  const [checked, setChecked] = useState(false)

  const updateCheck = () => {
    setChecked(!checked)
  }

  return (
    <Checkbox
      labelName={text('Label', 'Testing')}
      name="Test"
      value={true}
      onChange={updateCheck}
      checked={checked}
      feedback={select('Feedback', [undefined, 'error'])}
      error={text('Error', 'This field is required')}
    />
  )
}

Playground.story = {
  name: 'playground',
  parameters: { docs: { page: null, disable: true } },
}
