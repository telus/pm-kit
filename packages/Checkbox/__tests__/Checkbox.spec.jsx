import React from 'react'
import { render } from 'enzyme'

import Checkbox from '../Checkbox'

describe('Checkbox', () => {
  it('renders', () => {
    const checkbox = render(<Checkbox />)

    expect(checkbox).toMatchSnapshot()
  })
})
