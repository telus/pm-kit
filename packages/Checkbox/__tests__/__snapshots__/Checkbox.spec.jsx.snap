// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checkbox renders 1`] = `
.emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  margin: 0 0 0 52px;
  font-size: 14px;
}

.emotion-0 label {
  -webkit-letter-spacing: 0.05px;
  -moz-letter-spacing: 0.05px;
  -ms-letter-spacing: 0.05px;
  letter-spacing: 0.05px;
  margin-right: 0.5rem;
}

.emotion-1 {
  position: fixed;
  opacity: 0;
  pointer-events: none;
}

.emotion-6 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.emotion-5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 100% !important;
  height: 100% !important;
  left: 0 !important;
  top: 0 !important;
}

.emotion-3 {
  height: 15px;
  width: 15px;
  min-height: 20px;
  min-width: 20px;
  outline: 0;
  line-height: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  cursor: pointer;
  margin-top: 0.125rem;
  border: 2px solid #034045;
  border-radius: 6px;
}

.emotion-2 > svg path {
  stroke: white;
}

.emotion-4 {
  font-weight: bold;
  margin-left: 12px;
}

<div>
  <div
    class="emotion-0"
  />
  <input
    class="emotion-1"
    id="1"
    name=""
    type="checkbox"
    value=""
  />
  <label
    class="emotion-6"
    for="1"
  >
    <div
      class="emotion-5"
    >
      <span
        class="emotion-3"
      >
        <span
          class="emotion-2"
        >
          <svg
            height="17"
            width="20"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M18.031 1.866L7.154 13.442l-5.37-5.944"
              fill="none"
              fill-rule="evenodd"
              stroke="#034045"
              stroke-width="4"
            />
          </svg>
        </span>
      </span>
      <span
        class="emotion-4"
      />
    </div>
  </label>
</div>
`;
